webapp/channels/src/components/logged_in/logged_in.tsx

```diff
@@ -68,6 +72,11 @@ export default class LoggedIn extends React.PureComponent<Props> {
 
         this.updateTimeZone();
 
+        // Set up a timer to periodically update timezone
+        this.timezoneUpdateInterval = window.setInterval(() => {
```

Comment:
Only 1/5 and a bit nitpicky: To keep this more self-contained, the timezone handling logic for this could be pulled out into a helper component that renders null and just exists to contain all of the timezone update logic

Reply:
Ooh, that's a good suggestion, and something I haven't personally done enough of to have suggested in the first place. Will update!